packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx

```diff
@@ -0,0 +1,61 @@
+import { useCommonNavigate } from 'navigation/helpers';
+import React from 'react';
+import app from 'state';
+import { CWText } from '../../../components/component_kit/cw_text';
+import { CommunityPreviewCard } from './CommunityPreviewCard';
+import './TrendingCommunitiesPreview.scss';
+
+export const TrendingCommunitiesPreview = () => {
+  const navigate = useCommonNavigate();
+
+  const sortedCommunities = app.config.chains
+    .getAll()
+    .sort((a, b) => {
+      const threadCountA = app.recentActivity.getCommunityThreadCount(a.id);
+      const threadCountB = app.recentActivity.getCommunityThreadCount(b.id);
+      return threadCountB - threadCountA;
+    })
+    .map((community, i) => {
+      const monthlyThreadCount = app.recentActivity.getCommunityThreadCount(
+        community.id,
+      );
+      const isMember = app.roles.isMember({
+        account: app.user.activeAccount,
+        community: community.id,
+      });
+      const { unseenPosts } = app.user;
+      const hasVisitedCommunity = !!unseenPosts[community.id];
+
+      return (
+        <CommunityPreviewCard
+          key={i}
+          community={community}
+          monthlyThreadCount={monthlyThreadCount}
+          isCommunityMember={isMember}
+          hasUnseenPosts={app.isLoggedIn() && !hasVisitedCommunity}
+          onClick={() => {
+            navigate(`/${community.id}`);
+          }}
```

Comment:
can `sortedCommunities` return data instead of jsx? It's just separation of concerns

Reply:
yeah makes sense, will do that...